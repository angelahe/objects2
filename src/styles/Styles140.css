.AppContainer {
    background-color: #e6f5ff;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;

}

.AppPanel {
    background-color: whitesmoke;
    border-radius: 10px;
    border-width: 2px;
    border-color: #1aa3ff;
    margin: 10px;
    height: auto;
    width: 40%;
    justify-content: flex-start;

}
.Header {
    background-color: #1aa3ff;
    border-radius: 10px;
}

.AppBtn {
    color: white;
    height: 35px;
    width:35px;
    padding:0px;
    align-content: flex-start;
    border-radius: 5px;
    margin-left: 10px;

}
.ActionBox {
    height: auto;
    width: 100%;
    border-radius: 10px;

}
.ItemBox {
    border-radius: 10px;
    padding: 10px;
}
.AppHeader {
    background-color: #1aa3ff;
    display:flex;
    justify-content:space-between;
}

.AppList {
    width: 96%;
    padding: 5px;
}

.AppListItem {
    background-color: #e6f5ff;
}

.AppListItem:active {
    background-color: #ccebff;
    justify-content: space-around;
}

.AddItem {
    /*width: 50%;*/
    font-weight: bold;
    font-size: 1.5em;
    justify-content: flex-start;
    color: white;
    background-color: #1aa3ff;
    text-align: left;

}
.btnImg {
    height: 35px;
    width: 35px;
    background-color: #1aa3ff;
    margin:0;
}

.btnText {
    height: 35px;
    width: 200px;
    background-color: #1aa3ff;
    color: white;
    font-size: 1.5em;
    border-radius: 10px;

}

.DetailText {
    color: #1aa3ff;
    font-size: 1.5em;
    align-items: flex-start;
    justify-content: left;
    text-align: left;
    line-height: 1.5em;
}

.InputText {
    color: #1aa3ff;
    font-size: 1.5em;
}

.SummaryText {
    color: #1aa3ff;
    font-size: 1em;
    align-items: flex-start;
    text-align: left;
    margin-left: 10px;
    line-height: 1.5em;
}

hr {
    color: darkblue;
    height: 2px;
}

h2 {
    padding: 1px;
    color: #1aa3ff;
}

