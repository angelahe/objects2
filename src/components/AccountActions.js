import React from 'react';
import '../styles/Styles140.css';
import closebtn from '../images/close_FFFFFF.png';

const AccountActions = (props) => {

    function handleCloseClick() {
        props.closeClicked("actions");
    }


    function handleClick(e) {
        const amount = Number(document.getElementById("amount").value);
        switch (e.target.getAttribute("elem")) {
            case "deposit":
                props.deposit(amount, props.account);
                break;
            case "withdraw":
                props.withdraw(amount, props.account);
                break;
            default:
                //do nothing
        }
    }

    const accountName = props.account.acctName;
    const accountBalance = props.account.balance;
    return(
        <div>
            <div className = "ItemBox AppHeader">
                <span className = "AddItem">{accountName}</span>
                <span className = "AddItem">${accountBalance}</span>
                <button className="AppBtn" onClick={handleCloseClick}>
                    <img className="btnImg" src={closebtn} alt="Close"/>
                </button>
            </div>
            <div className = "ItemBox" onClick={handleClick}>
                <span className = "DetailText">For Account Name:</span>
                <span className = "DetailText" >{accountName}</span><br/><br/>
                <span className = "DetailText">Amount:</span>
                <input className = "InputText" id = "amount" placeholder="Enter an amount" type="number"></input><br/><br/>
                <button className="btnText" elem="deposit">Deposit</button>
                <button className="btnText" elem="withdraw">Withdraw</button>
            </div>
        </div>
    )
}
export default AccountActions;