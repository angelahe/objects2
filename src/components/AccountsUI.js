import React from 'react';
import Accounts from './accounts';
import Account from './account';
import AccountSummary from './AccountSummary';
import AccountCreate from './AccountCreate';
import AccountEdit from './AccountEdit';
import AccountDelete from './AccountDelete';
import AccountListItem from './AccountListItem';
import AccountActions from './AccountActions';
import addbtn from '../images/add_FFFFFF.png';

import '../styles/Accounts.css';


class AccountsUI extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            AccountList: new Accounts(),
            accountNumber: 0,
            nextId: 0,
            currentSelected: 0,
            currentAccount: "Savings",
            currentBalance: 0,
            lowestAccount: 0,
            highestAccount: 0,
            totalAccount: 0,
            addAccount: false,
            editShow: false,
            deleteShow: false,
            accountActionsShow: false
        }
    }

    onBtnAddClick = () => {
        const changeState = !this.state.addAccount;
        this.setState ({addAccount: changeState});
    }

    //make a copy of the current AccountList in state
    makeAccountCopy = () => {
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        return AccountListCopy;

    }

    onAccountCreate = (name, balance) => {
        const AccountListCopy = this.makeAccountCopy();
        AccountListCopy.addAccount(new Account(balance, name, this.state.nextId));

        this.setState({
            AccountList: AccountListCopy,
            accountNumber: AccountListCopy.Accounts.length,
            addAccount: !this.state.addAccount,
            nextId: this.state.nextId + 1
        });
    }

    onEditClick = (acctIndex) => {
        this.setState({
            editShow: !this.state.editShow,
            currentSelected: acctIndex
        });

    }

    onAccountUpdate = (name, balance, index) => {
        console.log("in account update with: ", name, balance, index);
        const updateWithThis = new Account(balance, name, index);
        const AccountListCopy = new Accounts();
        const changeState = !this.state.editShow;

        //make a copy of Accountlist, replacing the edited account
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            if(i === index)
                AccountListCopy.addAccount(updateWithThis);
            else
                AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }

        this.setState({
            AccountList: AccountListCopy,
            editShow: changeState
        });
    }

    onDeleteClick = (acctIndex) => {
        console.log("in delete click", acctIndex);
        this.setState({
            deleteShow: true,
            currentSelected: acctIndex
        });
    }

    onDeleteAccount = (acctIndex) => {
        console.log("in delete account", acctIndex);
        const AccountListCopy = new Accounts();

        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            if (i!== acctIndex)
                AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        this.setState({
            AccountList: AccountListCopy,
            deleteShow: false,
            accountNumber: AccountListCopy.Accounts.length,
            currentSelected: 0
        });
    }

    onAccountClick = (acctIndex) => {
        console.log("in account click", acctIndex);
        this.setState({
            accountActionsShow: true,
            currentSelected: acctIndex
        })
    }

    onDepositToAccount = (amount, acctIndex) => {
        console.log("in onDepositToAccount");
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        AccountListCopy.depositToAccount(acctIndex, amount);
        this.setState({
            AccountList: AccountListCopy,
            accountActionsShow: false,
        });
    }

    onWithdrawFromAccount = (amount, acctIndex) => {
        console.log("in onWithdrawFromAccount for of ", amount, acctIndex);
        const AccountListCopy = new Accounts();
        for(let i=0; i<this.state.AccountList.Accounts.length; i++) {
            if (i!== acctIndex)
                AccountListCopy.addAccount(this.state.AccountList.Accounts[i]);
        }
        AccountListCopy.withdrawFromAccount(acctIndex, amount);
        this.setState({
            AccountList: AccountListCopy,
            accountActionsShow: false
        });
    }

    render() {

//        console.log("Account list items are ", this.state.AccountList);
        console.log("addAccount is ", this.state.addAccount);
        const accountListItems = this.state.AccountList.Accounts.map((item, index) =>
           <AccountListItem key={item.acctId}
                            index = {index}
                            item={item}
                            accountClicked={this.onAccountClick}
                            editClicked={this.onEditClick}
                            deleteClicked={this.onDeleteClick}
           />
        );
//<AccountSummary accounts={this.state.AccountList.Accounts} />

        return(
            <div>
                <h1>Hello from Accounts</h1>
                <div className = "AccountContainer">
                    <AccountSummary accounts={this.state.AccountList} />
                    <div className = "AccountPanel">
                          <div className = "ItemBox AccountHeader">
                            <span className = "AddAccount">Add Account</span>
                            <button className="AccountBtn" onClick={this.onBtnAddClick}>
                                <img className="btnImg" src={addbtn} alt="Add"/>
            `                    </button>
                          </div>
                        <div className = "AccountList">
                            {accountListItems}
                        </div>
                    </div>
                    <div className = "AccountPanel">
                        {(this.state.addAccount)
                            ? <AccountCreate {...this.state}
                                             createClicked = {this.onAccountCreate}
                              />
                            : null }
                        {(this.state.editShow)
                            ? <AccountEdit {...this.state}
                                         index= {this.state.currentSelected}
                                         updateAccount = {this.onAccountUpdate}
                              />
                            :null }
                        {(this.state.deleteShow)
                            ? <AccountDelete {...this.state}
                                           index = {this.state.currentSelected}
                                           deleteAccount = {this.onDeleteAccount}
                              />
                            :null }
                        {(this.state.accountActionsShow)
                            ? <AccountActions {...this.state}
                                              index={this.state.currentSelected}
                                              deposit={this.onDepositToAccount}
                                              withdraw={this.onWithdrawFromAccount}
                            />
                            : null
                        }
                    </div>

                </div>

            </div>
        )

    }
}
export default AccountsUI