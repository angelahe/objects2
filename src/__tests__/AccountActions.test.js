import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import AccountActions from "../components/AccountActions";
import Accounts from "../components/accounts";
import '../styles/Styles140.css'
import renderer from "react-test-renderer";
import AccountsUI from "../components/AccountsUI";

const props = new AccountsUI();
const account = {acctId : 1,
  acctName: "Savings",
  balance: 1000};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AccountActions account={account}
                                  deposit={props.onDepositToAccount}
                                  withdraw={props.onWithdrawFromAccount}
                  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<AccountActions account={account}
                          deposit={props.onDepositToAccount}
                          withdraw={props.onWithdrawFromAccount}
          />);
});

test('AccountUI renders correctly', () => {
  const tree = renderer.create(<AccountActions account={account}
                                               deposit={props.onDepositToAccount}
                                               withdraw={props.onWithdrawFromAccount}
                               />).toJSON();
  expect(tree).toMatchSnapshot();
});