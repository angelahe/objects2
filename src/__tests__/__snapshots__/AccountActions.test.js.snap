// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUI renders correctly 1`] = `
<div>
  <div
    className="ItemBox AppHeader"
  >
    <span
      className="AddItem"
    >
      Savings
    </span>
    <span
      className="AddItem"
    >
      $
      1000
    </span>
  </div>
  <div
    className="ItemBox"
    onClick={[Function]}
  >
    <span
      className="DetailText"
    >
      For Account Name:
    </span>
    <span
      className="DetailText"
    >
      Savings
    </span>
    <br />
    <br />
    <span
      className="DetailText"
    >
      Amount:
    </span>
    <input
      className="InputText"
      id="amount"
      placeholder="Enter an amount"
      type="number"
    />
    <br />
    <br />
    <button
      className="btnText"
      elem="deposit"
    >
      Deposit
    </button>
    <button
      className="btnText"
      elem="withdraw"
    >
      Withdraw
    </button>
  </div>
</div>
`;
